Source/Scene/ShadowMap.js

```diff
@@ -307,110 +455,292 @@ define([
-                appearance : new PerInstanceColorAppearance({
-                    translucent : false,
-                    flat : true
-                }),
-                asynchronous : false
-            });
+    var debugCascadeColors = [Color.RED, Color.GREEN, Color.BLUE, Color.MAGENTA];
 
-            //// Create the camera frustum
-            //var nearPlane = frameState.camera.frustum.near;
-            //var farPlane = frameState.camera.frustum.far;
-            //shadowMap._debugCameraFrustum = new Primitive({
-            //    geometryInstances : new GeometryInstance({
-            //        geometry : new BoxOutlineGeometry({
-            //            minimum : new Cartesian3(-0.0, -0.0, -nearPlane),
-            //            maximum : new Cartesian3(0.0, 0.0, -farPlane)
-            //        }),
-            //        attributes : {
-            //            color : ColorGeometryInstanceAttribute.fromColor(Color.RED)
-            //        }
-            //    }),
-            //    appearance : new PerInstanceColorAppearance({
-            //        translucent : false,
-            //        flat : true
-            //    }),
-            //    asynchronous : false
-            //});
+    function applyDebugSettings(shadowMap, frameState) {
+        if (!shadowMap.debugShow) {
+            return;
         }
 
-        //if (!shadowMap.debugFreezeFrame) {
-        //
-        //    //var nearPlane = frameState.camera.frustum.near;
-        //    //var point = Matrix4.multiplyByPoint(frameState.context.uniformState.inverseViewProjection, new Cartesian3(0, 0, -nearPlane), new Cartesian3());
-        //    //console.log(point.x, point.y, point.z);
-        //    //console.log(frameState.camera.positionWC.x, frameState.camera.positionWC.y, frameState.camera.positionWC.z);
-        //    //
-        //    //// TODO : possibly get this from the camera instead
-        //    //var cameraMatrix = frameState.context.uniformState.inverseViewProjection;
-        //    //shadowMap._debugCameraFrustum.modelMatrix = cameraMatrix;
-        //}
-
+        shadowMap._debugLightFrustum = shadowMap._debugLightFrustum || createDebugFrustum(Color.YELLOW);
+        Matrix4.inverse(shadowMap._shadowMapCamera.getViewProjection(), shadowMap._debugLightFrustum.modelMatrix);
         shadowMap._debugLightFrustum.update(frameState);
-        shadowMap._debugLight.update(frameState);
-        //shadowMap._debugCameraFrustum.update(frameState);
+
+        for (var i = 0; i < shadowMap._numberOfCascades; ++i) {
+            shadowMap._debugCascadeFrustums[i] = shadowMap._debugCascadeFrustums[i] || createDebugFrustum(debugCascadeColors[i]);
+            Matrix4.inverse(shadowMap._cascadeCameras[i].getViewProjection(), shadowMap._debugCascadeFrustums[i].modelMatrix);
+            shadowMap._debugCascadeFrustums[i].update(frameState);
+        }
 
         updateDebugShadowViewCommand(shadowMap, frameState);
     }
 
+    function ShadowMapCamera() {
```

Comment:
The `Camera` used to be bare bones like this. It just had properties and made sure the view and projection matrices were computed only once a frame after all of the camera updates. Then, we added a bunch of convenience functions. It might not seem like it now, but you might want to switch to using a `Camera`. It would be cool to have a point light attached to a camera animated over a flight, illuminate a rectangle on the globe, etc.